### PR TITLE
Generate changelog and add PR template

### PR DESCRIPTION
**Pull request by JordanMartinez**
Part of purescript/purescript#3984

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Changelog
+
+Notable changes to this project are documented in this file. The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+Breaking changes (😱!!!):
```
Comment (thomashoneyman) [inline CHANGELOG.md]: We're removing this, right?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+Clearly and concisely describe the purpose of the pull request. If this PR relates to an existing issue or change proposal, please link to it. Include any other background context that would help reviewers understand the motivation for this PR.
+
+---
+
+**Checklist:**
+
+- [ ] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
+- [ ] Linked any existing issues or proposals that this pull request should close
+- [ ] Updated or added relevant documentation in the README and/or documentation directory
```
Comment (thomashoneyman) [inline .github/PULL_REQUEST_TEMPLATE.md]: Perhaps we should remove "in the README and/or documentation directory" for core libraries and just leave the documentation note.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+**Description of the change**
+
+Clearly and concisely describe the purpose of the pull request. If this PR relates to an existing issue or change proposal, please link to it. Include any other background context that would help reviewers understand the motivation for this PR.
+
+---
+
+**Checklist:**
+
+- [ ] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
```
Comment (thomashoneyman) [inline .github/PULL_REQUEST_TEMPLATE.md]: I agree with @f-f that perhaps we don't require a _link_ to this information, but rather just the PR number -- like `Updated return type of function to be an integer (#55)`. We may not need the username either.

File: CHANGELOG.md (hunk 1 above)
Comment (JordanMartinez) [reply CHANGELOG.md]: Yeah, I forgot to remove it from my template before submitting a few test PRs.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (JordanMartinez) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'll make that change to my template.

Comment (JordanMartinez) [reply .github/PULL_REQUEST_TEMPLATE.md]: How's this?

```
- [ ] Added the change to the changelog's "Unreleased" section with a 
      reference to this PR (e.g. "- Made a change (#22)")
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (JordanMartinez) [reply .github/PULL_REQUEST_TEMPLATE.md]: Sounds good to me.

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**